content/src/logic/snapshots.ts

```diff
@@ -0,0 +1,107 @@
+import { createFileWriter, IFile } from '../ports/fileWriter'
+import { AppComponents } from '../types'
+import {
+  deleteSnapshotsInTimeRange,
+  findSnapshotsStrictlyContainedInTimeRange,
+  getSnapshotHashesNotInTimeRange,
+  saveSnapshot,
+  streamActiveDeploymentsInTimeRange
+} from './database-queries/snapshots-queries'
+import { divideTimeInYearsMonthsWeeksAndDays, intervalSizeLabel, isTimeRangeCoveredBy, TimeRange } from './time-range'
+
+export type NewSnapshotMetadata = {
+  hash: string
+  timeRange: TimeRange
+  numberOfEntities: number
+  replacedSnapshotHashes?: string[]
+}
+
+export async function generateAndStoreSnapshot(
```

Comment:
the format of the old snapshot and new files is the same, right?

Reply:
yes